project/src/js/ui/round/view/roundView.js

```diff
@@ -375,20 +375,30 @@ function renderGameEndedActions(ctrl) {
     m('strong', result), m('br')
   ];
   resultDom.push(m('em.resultStatus', status));
-  const buttons = ctrl.data.player.spectator ? [
-    button.shareLink(ctrl),
-    button.sharePGN(ctrl),
-    button.analysisBoard(ctrl),
-    ctrl.data.tv ? tvChannelSelector(ctrl) : null
-  ] : [
-    button.shareLink(ctrl),
-    button.sharePGN(ctrl),
-    button.newOpponent(ctrl),
-    button.answerOpponentRematch(ctrl),
-    button.cancelRematch(ctrl),
-    button.rematch(ctrl),
-    button.analysisBoard(ctrl)
-  ];
+  let buttons = null;
+  if (ctrl.data.tournament) {
```

Comment:
What if you are spectator of a tournament? I think you should have also the regular spectator buttons.

Same for player actually: the newOpponent and all rematch related buttons should not be displayed in a tournament context ; it's not the same for the other buttons that should remain.

So the logic must be changed to take this into account.